src/libs/ReportActionsUtils.ts

```diff
@@ -39,202 +46,142 @@ Onyx.connect({
-            // First sort by timestamp
-            if (first.created !== second.created) {
-                return (first.created < second.created ? -1 : 1) * invertedMultiplier;
-            }
-
-            // Then by action type, ensuring that `CREATED` actions always come first if they have the same timestamp as another action type
-            if ((first.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED || second.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED) && first.actionName !== second.actionName) {
-                return (first.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED ? -1 : 1) * invertedMultiplier;
-            }
-            // Ensure that `REPORTPREVIEW` actions always come after if they have the same timestamp as another action type
-            if ((first.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW || second.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW) && first.actionName !== second.actionName) {
-                return (first.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW ? 1 : -1) * invertedMultiplier;
-            }
-
-            // Then fallback on reportActionID as the final sorting criteria. It is a random number,
-            // but using this will ensure that the order of reportActions with the same created time and action type
-            // will be consistent across all users and devices
-            return (first.reportActionID < second.reportActionID ? -1 : 1) * invertedMultiplier;
-        })
-        .value();
+
+    return reportActions.filter(Boolean).sort((first, second) => {
+        // First sort by timestamp
+        if (first.created !== second.created) {
+            return (first.created < second.created ? -1 : 1) * invertedMultiplier;
+        }
+
+        // Then by action type, ensuring that `CREATED` actions always come first if they have the same timestamp as another action type
+        if ((first.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED || second.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED) && first.actionName !== second.actionName) {
+            return (first.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED ? -1 : 1) * invertedMultiplier;
+        }
+        // Ensure that `REPORTPREVIEW` actions always come after if they have the same timestamp as another action type
+        if ((first.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW || second.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW) && first.actionName !== second.actionName) {
+            return (first.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW ? 1 : -1) * invertedMultiplier;
+        }
+
+        // Then fallback on reportActionID as the final sorting criteria. It is a random number,
+        // but using this will ensure that the order of reportActions with the same created time and action type
+        // will be consistent across all users and devices
+        return (first.reportActionID < second.reportActionID ? -1 : 1) * invertedMultiplier;
```

Comment:
Seems like this must be returning an object instead of an array 🤔 based on the stacktrace shared here https://github.com/Expensify/App/issues/29965#issuecomment-1770795985

Reply:
Yep you are right, this should have gone to the `getMostRecentIOURequestActionID` method